source/common/http/utility.h

```diff
@@ -158,7 +161,8 @@ class PercentEncoding {
 
 private:
   // Encodes string view to its percent encoded representation, with start index.
-  static std::string encode(absl::string_view value, const size_t index);
+  static std::string encode(absl::string_view value, const size_t index,
+                            const absl::flat_hash_set<char>& reserved_char_set);
```

Comment:
Are there tests or uses of this function?  If not, we should delete it.

Reply:
Nevermind, this helper function is used internally by encode, which does a scan first.